src/arrayMethodSort.js

```diff
@@ -4,8 +4,32 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let switcher = true;
+
+    while (switcher) {
+      switcher = false;
+
+      for (let i = 0; i < this.length - 1; i++) {
+        const firstValue = this[i];
+        const secondValue = this[i + 1];
+        const compareString = (a, b) => (String(a) > String(b) ? 1 : -1);
+
+        const callback =
+          compareFunction === undefined
+            ? compareString(firstValue, secondValue)
+            : compareFunction(firstValue, secondValue);
+
+        if (callback > 0) {
+          this[i] = secondValue;
+          this[i + 1] = firstValue;
+          switcher = true;
+          break;
```

Comment:
The `break` statement inside the if block will cause the loop to terminate after the first swap, which is not how a bubble sort should work. Bubble sort requires the loop to continue until no swaps are needed. Remove the `break` statement to ensure the loop continues to iterate over the entire array.